### ISSUE TITLE
Other languages

### ISSUE THREAD

**Issue by EmreOzkose** [gmltmd789/UnitSpeech#6]
Hi, did you try train a different language for voice conversion from pretrained models ? Can you give some hints for this issue, which modules should be re-trained ?

**Comment by gmltmd789** [comment]
Hi,

If you wish to implement other languages in the current version, you will need to restart the process outlined in the paper from the beginning.

Assuming you want to expand to Korean:

1. First, train a backbone TTS model using Korean multi-speaker data. As in the paper, the encoder should be speaker-independent, with the speaker condition only entering the decoder. For the speaker condition, you can use an existing speaker encoder or WavLM, which we used.

If you are curious about the training method, we followed the Grad-TTS training approach. The main difference is that while Grad-TTS used a diffusion prior for the encoder output, we used a zero mean prior.

2. Then, train a unit encoder that can replace the text encoder for voice conversion and fine-tuning using untranscribed speech. Essentially, units are extracted using HuBERT and the k-means clustering algorithm. For extraction, you can refer to the usage of unit_extractor in the repository's scripts/finetune.py.

The unit encoder directly replaces the text encoder and is trained alongside the decoder as part of the TTS model training objective. During this phase, only the unit encoder is trained while the decoder is frozen.

Afterwards, you can follow the fine-tuning and inference steps as in our repository. However, note that while units are suitable for fine-tuning, they may lead to pronunciation loss when used for actual voice conversion. Therefore, if you seek to improve pronunciation in voice conversion, we recommend training a contentvec encoder, with the same method using for unit encoder. 
Contentvec is a self-supervised representation that contains more pronunciation information than units.

**Comment by EmreOzkose** [comment]
thank you for detailed explanation.

**Comment by adrianstanea** [comment]
Sorry to reopen this issue.

I'm also trying to train the models from scratch and have some issues.

Step 1: training backbone TTS model
* How would the process of extracting speaker embeddings look like? Do get an average embedding with all the audios for a speaker or re-compute with each wav?. Also should we always normalize spk_emb to unit length like this: `spk_emb = spk_emb / spk_emb.norm()`
* I see that in the last change you used mel-spectrograms scaled to [-1 1]. Should the train process also do the same operation on the samples of the dataset?

Step 2: Unit Encoder
* In the fine_tune.py script I see that we don use the Duration Predictor.  Can i keep a similar logic as in step 1 but just use the scaled version of (unit, duration) using the `processs_unit` function?
